lib/reporters/html/partials/scripts.js

```diff
@@ -0,0 +1,89 @@
+'use strict';
+
+exports.scripts = () => {
+
+    return `<script>
+  headings = [];
+
+  onload = function () {
+      headings = document.querySelectorAll('h2');
+      reset();
+  };
+
+  onscroll = function (e) {
+      var heading = find(window.scrollY);
+      if (!heading) {
+        var current = document.querySelector('#menu a.active');
+        if (current) {
+          current.className = '';
+        }
+        return;
+      }
+      var links = document.querySelectorAll('#menu a')
+              , link;
+
+      for (var i = 0, len = links.length; i < len; ++i) {
+          link = links[i];
+          link.className = link.getAttribute('href') == '#' + heading.id ? 'active' : '';
+      }
+  };
+
+  function find (y) {
+      var i = headings.length
+              , heading;
+
+      while (i--) {
```

Comment:
I believe this would exit when `i === 0`, does headings[0] not need to be processed?

Reply:
0 does get processed. The while Evals then the -1 is done